music21/stream/base.py

```diff
@@ -434,13 +434,17 @@ def __getitem__(self, k: slice) -> t.List[M21ObjType]:
     @overload
     def __getitem__(
         self,
-        k: t.Type[ChangedM21ObjType]
+        k: t.Union[t.Type[ChangedM21ObjType], t.Collection[t.Type[ChangedM21ObjType]]]
     ) -> iterator.RecursiveIterator[ChangedM21ObjType]:
```

Comment:
does this properly type the output for mypy?  (just want to be sure that that was checked)

Reply:
I think I'm not quite following -- can you explain what you mean? (`mypy` passes, and I didn't touch the return type)

Reply:
What I mean is something like, say you start typing:

```
for el in s[note.Note, chord.Chord]:
    print(el.pitches)
    print(el.recurse())
```
will the `el.pitches` line start to auto-suggest .pitches when you've typed .pi... (like does it know that all possible members have .pitches in it), and will mypy choke on the `el.recurse()` -- knowing that the union of note.Note and chord.Chord does not have `.recurse()` defined on all members -- or even one member.

And if you do:

```
for el in s[note.Note, chord.Chord]:
    if not isinstance(el, note.Note):
         print(el.closedPosition())
```
and know that this is fine?  I'm just wondering if the typing is inferred from the ChangedM21Type.

And most importantly, does this still raise a mypy error:

```
for el in s[note.Note]:
     print(el.isMajorTriad())
```

while this passes?

```
for el in s[note.Note]:
     print(el.pitch)
```

I'll just grab the PR and play with it.  But I'm wondering if putting the two different types of things in the same `@overload` might cause problems.